src/appsettings/settings.py

```diff
@@ -837,6 +879,27 @@ def __init__(
         if max_length is not None:
             self.validators.append(MaxLengthValidator(max_length))
 
+    def decode_environ(self, value):
+        """
+        Decode JSON value or split value by a delimiter to a list, if JSONDecodeError is raised.
+
+        Default delimiter is a colon, can be changed via attribute ``delimiter``.
+
+        Args:
+            value (str):
+
+        Returns:
+            Iterable:
+        """
+        try:
+            value = super(IterableSetting, self).decode_environ(value)
+        except json.decoder.JSONDecodeError:
+            split_value = value.split(self.delimiter)
+            item_convert_func = self.item_type or (lambda v: v)
+            value = [item_convert_func(v) for v in split_value]
+        finally:
```

Comment:
I'd drop the `finally` statements.

Reply:
Is there a reason or effect? Anyway, done.

Reply:
I believe it may have triggered error due to undefined `value` in case of exception in the `except` branch.